tutorial_pl/README.md

```diff
@@ -0,0 +1,152 @@
+
+Rozwiązania powinny wczytywać ze standardowego wejścia (np. cin) i wypisywać na standardowe wyjście (np. cout). Bardzo zalecane jest umieszczenie komentarza na samej górze każdego kodu z opisem rozwiązania (czy jest poprawne, czy nie - jak nie to dlaczego nie), opisem złożoności i autorem rozwiązania.
+
+
+# Generatorka
+
+By wygenerować testy, potrzebna jest generatorka. Program ma przyjmować argumenty i na standardowe wyjście (np. cout) wypisać test. Za pomocą argumentów określa się rozmiar testu, itp. Przykładowa generatorka:
+
+``` cpp
+#include <bits/stdc++.h>
+using namespace std;
+using L = long long;
+
+mt19937_64 engine(chrono::system_clock::now().time_since_epoch().count());
+// losowy int z przedziału [a, b]: (więcej informacji na en.cppreference.com/w/cpp/numeric/random )
+L rd(L a, L b) {
+	return uniform_int_distribution<L>(a, b)(engine);
+}
+
+int main(int argc, char* argv[]) {
+	// argument 0 jest pomijany
+	constexpr int ilosc_argumentow = 2;
+	assert(argc == ilosc_argumentow + 1);
+
+	// zamienianie z tablicy charów na liczbę (więcej informacji na en.cppreference.com/w/cpp/string/byte/atoi )
+	int max_n = atoi(argv[1]),
+	    max_val = atoi(argv[2]);
+
+	int n = rd(1, max_n);
+	cout << n << '\n';
+	for(int i = 0; i < n; ++i)
+		cout << rd(-max_val, max_val) << ' ';
+}
+```
+
+By ją uruchomić, należy skompilować ją normalnie i odpalić `./generatorka 5 10` (czyli argumentami do generatorki są liczby 5, a potem 10).
+
+# Generowanie testów
+
+Ustal jakie będą grupy testów, ile będzie testów w danej grupie i jakie będą ich wielkości. Przydziałem punktów do każdej grupy testów zajmiemy się później. Każdy plik wejściowy/wyjściowy zazwyczaj jest nazywany w sposób tag1a, gdzie '1' to numer grupy, a 'a' to numer testu w grupie. Testy przykładowe mają mieć numer grupy 0. Użytkownik widzi status swojego kodu na testach przykładowych.
+
+Sip potrafi automatycznie generować testy za pomocą pliku Sipfile. Przykładowy plik Sipfile:
+```
+# czas (sekund), po którym przerywa wzorcówkę podczas generowania testów:
+default_time_limit: 15
+static: [
+	# Tu umieść testy zrobione ręcznie (testy przykładowe)
+	# Syntax: <test-range>
+	tag0a
+]
+gen: [
+	# Tu umieść zasady do generowania testów
+	# Syntax: <test-range> <generator> [generator arguments]
+	tag1a-f utils/generatorka 5 10
+	tag2a utils/generatorka 100 100000
+	tag2b utils/generatorka 1000 10000
+	tag3-5a-d utils/generatorka 100000 1000000000
```

Comment:
Wydaje mi się, że w podanej formie sim nie zaakceptuje tego test range'a, ale w tej poprawionej tak.
```suggestion
	tag3a-5d utils/generatorka 100000 1000000000
```

Reply:
Też można dodać info, że testy nie muszą mieć numeru testu w grupie i dać na to test range'a np. `tag6-8 ...`.